include/votca/xtp/davidsonsolver.h

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2009-2019 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef __VOTCA_TOOLS_DAVIDSON_SOLVER_H
+#define __VOTCA_TOOLS_DAVIDSON_SOLVER_H
+
+#include <iostream>
+#include <stdexcept>
+#include <chrono>
+
+#include <votca/xtp/eigen.h>
+
+#include <votca/xtp/orbitals.h>
+#include <votca/xtp/qmstate.h>
+#include <votca/xtp/rpa.h>
+#include <votca/xtp/threecenter.h>
+
+using boost::format;
+using std::flush;
+
+namespace votca {
+namespace xtp {
+
+/**
+* \brief Use Davidson algorithm to solve A*V=E*V
+
+**/
+
+class DavidsonSolver {
+
+ public:
+  DavidsonSolver(ctp::Logger &log);
+
```

Comment:
what happens if the solver does not converge, what is the max num of iterations? Errorhandling ?

Reply:
Eigen has this .info() method and uses an enum for the error codes.

Reply:
The code throw now an error if Davidson doesn't converge. The max number of iteration is defined in `davidosn-solver.h` and is 1000, which is a bit much maybe. We will use the eigen .info() when/if we'll move routines in Eigen.

Reply:
okay